04_health/exposure-to-trauma/exposure-to-trauma-2020.Rmd

```diff
@@ -0,0 +1,434 @@
+  
+```
+
+269 counties are missing an injury death rate value in 2020. This is roughly the same number of missing values in previous years.
+```{r, message = FALSE}
+deaths %>%
+  group_by(year) %>%
+  summarize(
+    missing = sum(is.na(rate_injury_death))
+  )
+
+```
+
+This table shows the injury death rates in each county. Click on the variable columns to sort the table. 
+    todo(): could not get `reactable()` to work, but it worked for previous version of this metric
+```{r table, echo = FALSE}
+# deaths %>%
+#   select(
+#     county_name, GEOID, year,
+#     rate_injury_death
+#   ) %>%
+#   reactable(
+#     filterable = TRUE,
+#     searchable = TRUE,
+#     defaultPageSize = 10,
+#     highlight = TRUE
+#   )
+
+```
+
+
+## 3. Add data quality flags
+This section creates a variable for the coefficient of variation (CV) for each county and assigns a quality indicator based on that county's CV. Smaller CVs are indicative of high quality data.
+
+```{r}
+deaths <- deaths %>%
+  mutate(cv = rate_injury_death_se / rate_injury_death,
+         rate_injury_death_quality = case_when(
+           cv < 0.2 ~ 1,
+           cv < 0.4 ~ 2,
+           cv >= 0.4 ~ 3
+         )
+  )
+
+```
+
+Most estimates have small coefficients of variation and are therefore of high quality.
+```{r}
+deaths %>%
+  ggplot(aes(x = rate_injury_death_quality)) +
+  geom_bar()
+  
+```
+
+Most estimates have small standard errors.
+```{r}
+deaths %>%
+  ggplot(aes(rate_injury_death, rate_injury_death_se)) +
+  geom_point(alpha = 0.1, size = 0.5) +
+  geom_abline(aes(slope = 0.4, intercept = 0)) +
+  labs(title = "Most Estimates Have Small SEs",
+       subtitle = "Line represents a CV of 0.4") +
+  coord_equal() +
+  scatter_grid()
+
+```
+
+
+## 4. Validation
+This section checks our results against the same metric created by [County Health Rankings](https://www.countyhealthrankings.org/explore-health-rankings/county-health-rankings-model/health-factors/social-economic-factors/community-safety/injury-deaths?year=2023). We use 2022 CHR data, because they also used 2016-2020 data from CDC WONDER in that release ([source](https://www.countyhealthrankings.org/sites/default/files/media/document/2022%20Analytic%20Documentation.pdf)). This section also creates two visualizations to check the validity of our data.
+
+### Download, read in, and clean the 2022 County Health Rankings Data
+The landing page to download these data is [here](https://www.countyhealthrankings.org/health-data/methodology-and-sources/rankings-data-documentation).
+
+```{r}
+# Download file directly from internet
+download.file(
+  url = "https://www.countyhealthrankings.org/sites/default/files/media/document/2022%20County%20Health%20Rankings%20Data%20-%20v1.xlsx",
+  destfile = here::here("04_health/exposure-to-trauma/data/chr2022.xlsx"),
+  mode = "wb"
+)
+
+# Read in CHR data
+chr2022 <- readxl::read_excel(here::here("04_health", "exposure-to-trauma", "data", "chr2022.xlsx"),
```

Comment:
Setting `guess_max = Inf` produced a different warning: `Warning: `guess_max` is a very large value, setting to `21474836` to avoid exhausting memoryNew names:` - but I still think this is better than the previous warning which was `Warning: Expecting logical in Q3172 / R3172C17: got 'x'New names:`.